src/cpp/session/modules/SessionFind.cpp

```diff
@@ -470,6 +470,12 @@ class GrepOperation : public boost::enable_shared_from_this<GrepOperation>
       }
    }
 
+   Error testWritePermissions(const FilePath& filePath)
+   {
+      std::shared_ptr<std::ostream> testStream;
+      return filePath.openForWrite(testStream, /*in_truncate*/ false);
+   }
+
```

Comment:
I'm not sure if there's a general use case for a test like this. If there is it would be good to have such a function in FilePath.hpp/cpp. What do you think @jmcphers?

Reply:
@jmcphers There is a function in FileMode.hpp that does essentially the same thing `isFileWriteable` but doesn't work for Windows (in general, FileMode.hpp isn't supported on Windows). Where that function is called it looks like we don't do any type of check for Windows so it might make sense to add this somewhere more generic.

Reply:
There is a subtle distinction between "does this file have the write access bit set for me" and "can I actually write to this file". There is no way to validate the latter without actually opening a handle which can have some side effects (especially on NFS mounted volumes). I think it's fine to move this into FilePath itself but we might leave a comment about reading the access bits instead unless it's important to guarantee success.

Reply:
I added this to FilePath, in the same commit I fixed a bug I discovered in testing this where we were losing the detailed error message in certain cases.